packages/drupal/custom/custom.module

```diff
@@ -180,3 +181,14 @@ function _custom_key_auth_form_access(UserInterface $user): AccessResult {
   $access->addCacheableDependency($user);
   return $access;
 }
+
+/**
+ * Implements hook_file_download().
+ *
+ * Prevent any access to keys.
+ */
+function custom_file_download($uri) {
+  if (str_starts_with($uri, 'private://') && str_ends_with($uri, '.key')) {
```

Comment:
I think `private://.key?foo=bar` can trick this check.

Reply:
In `hook_file_download($uri)`, `$uri` is not sensitive to query string so should be fine.